### PR TITLE
[No QA] update PR checklist

### PR DESCRIPTION
**Pull request by marcochavezf**
### Details

This updates the PR checklist for both Contributors and Reviewers based on the recent changes to this [doc](https://docs.google.com/document/d/1_lKe5UcSAbLOBsf8c7qmTHfE-JZF6PztLtjP3-7LhfE/edit).  
[slack convo](https://expensify.slack.com/archives/C01SKUP7QR0/p1647374658021899?thread_ts=1643767911.674959&cid=C01SKUP7QR0)

### Fixed Issues

N/A

### Tests

N/A

### QA Steps

N/A

### Screenshots

N/A

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (aldo-expensify) [inline .github/PULL_REQUEST_TEMPLATE.md:55]: I think the indentation is off here

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (aldo-expensify) [inline .github/PULL_REQUEST_TEMPLATE.md:122]: These two instructions are not clear to me. Am I supposed to only bind to "this" if we use "this" inside the internal method? or always?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (aldo-expensify) [inline .github/PULL_REQUEST_TEMPLATE.md:112]: While DRYness is good, I think it can be a trade off decision sometimes between not repeating and making the code simple. Being too dry can sometimes result in very abstract code. How can someone say that it has been "verified" for in terms of DRY?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (aldo-expensify) [inline .github/PULL_REQUEST_TEMPLATE.md:113]: Similar to DRY comment, I think having stuff in CONST is good, but doesn't apply always, and I think this doesn't define very well when it is a good idea to put something there.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (aldo-expensify) [inline .github/PULL_REQUEST_TEMPLATE.md:109]: Do we need more than one Expensify engineer to agree?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (aldo-expensify) [inline .github/PULL_REQUEST_TEMPLATE.md:91]: Is the offline behaviour always going to match the expected behaviour? do we need a new section in the GH issue to describe how it should work when offline if it is different?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (aldo-expensify) [inline .github/PULL_REQUEST_TEMPLATE.md:75]: These two instructions are not clear to me. Am I supposed to only bind to "this" if we use "this" inside the internal method? or always?

**Review by aldo-expensify — COMMENTED** [review]
Left some comments / questions

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (aldo-expensify) [reply .github/PULL_REQUEST_TEMPLATE.md:113]: Does it apply anytime we have some hardcoded value (string or number)?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (aldo-expensify) [reply .github/PULL_REQUEST_TEMPLATE.md:109]: I don't think it will be easy to know if there is a "new code pattern" added, unless you know really well the code base.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (marcochavezf) [reply .github/PULL_REQUEST_TEMPLATE.md:55]: ah yeah this is on purpose because this is a sub-bullet point of `I followed proper code patterns`

Comment (aldo-expensify) [reply .github/PULL_REQUEST_TEMPLATE.md:55]: image

**Comment by marcochavezf** [comment]
> Left some comments / questions

Thanks Aldo! 😀 Could you add the questions related to the checklist points directly in the [gdoc](https://docs.google.com/document/d/1_lKe5UcSAbLOBsf8c7qmTHfE-JZF6PztLtjP3-7LhfE/edit)?

**Comment by aldo-expensify** [comment]
Moved my dumb questions to the doc :), I'll mark them as resolved here.

Meanwhile, should we wait for some feedback before approving? Or should I just approved and if my questions translate in any change, we do a new PR?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (marcochavezf) [reply .github/PULL_REQUEST_TEMPLATE.md:55]: Ahh I see, thanks for pointing that, fixing it

**Comment by aldo-expensify** [comment]
> This is an ongoing iterative process, so we'll be updating the PR template when there are new suggestions to the gdoc, meanwhile, I think it's fine to merge this PR and create a new one when the questions are resolved in the doc

Sounds good, thanks!

**Review by aldo-expensify — APPROVED** [review]

**Pull request merged**
